### DISCUSSION TITLE
Esense sensor

### DISCUSSION THREAD

**Discussion by stephencrowther** [CJNE/ha-myenergi#608] [Q&A] [answered]
I’m assuming the status of esense (on/off) isn’t anything that’s currently available in the API and is the reason there isn’t a sensor for it?

I think it would be useful to have a sensor for this so you can use it to control devices through HA based o whether you are on an off peak period ie economy 7 or not.

**Comment by G6EJD** [comment, marked as answer]
But there is no API control or reading of the eSense state so this can never be implemented until MyEnergi bring out new products that may introduce the capabilities and assuming then that someone reverse engineers the interface and makes the necessary integration changes, with that in mind, I can't see this ever being feasible.

**Comment by stephencrowther** [comment]
Yeh that’s what I suspected but was just checking.

shame.

**Comment by stephencrowther** [comment]
Just thought of something actually. The relays can be set to trigger when esense comes on I think and they are exposed in HA. So this will do essentially the same thing as what I was looking for. Just a shame a physical relay has to switch to do this (relays wear out so best not to over use them really)

**Comment by G6EJD** [comment]
A relay's electrical life is typically 1.3 million operations, and it's mechanical life is up to 100 million operations, so there's no chance of the relays being worn-out!

If the relays are used inside their design rating of 16Amps there will be no issues.

**Comment by stephencrowther** [comment]
Not by my experience. I’ve had relays fail on other devices that haven’t lasted anywhere near that long.

**Reply by G6EJD** [reply]
Then they were incorrectly rated or of poor quality.

**Comment by stephencrowther** [comment]
Having said that I’ve defeated my own argument because even they did go faulty or wear out it’s an easily replaceable separate board.

**Comment by stephencrowther** [comment]
That’s a maybe. But saying they will never wear out is plainly wrong. How do you know the quality of the ones used in eddi? Did you design it?

**Reply by G6EJD** [reply]
You're overthinking the problem. MyEnergi use Relpol relays rated for  an electrical life expectancy of typically at least 100,000 operations.

That's 28 operations per day for 10-years and no Eddi is ever going to exceed that.
